components/core/src/error.rs

```diff
@@ -501,6 +511,18 @@ impl error::Error for Error {
     }
 }
 
+impl From<env::JoinPathsError> for Error {
+    fn from(err: env::JoinPathsError) -> Self {
+        Error::JoinPathsError(err)
+    }
+}
+
+impl From<ffi::OsString> for Error {
```

Comment:
I'd avoid general conversions like this. They're pretty hard to track down as more and more people leverage them. We see this with the `io::Error` From implementation - it lacks the context of *where* and *why* this error occurred.

Instead I'd recommend an error such as "Error::BadEnvironmentString" or something related to what was being read at the time of failure.